### PR TITLE
Add serialization to Report type

### LINKED ISSUE #4: Add serialization for Report
This allows a Report to be used as part of a quote request to Intel's AESM daemon.

### PR DESCRIPTION
**Pull request by lkatalin**
Signed-off-by: Lily Sturmann <lsturman@redhat.com>

This is necessary in order to use the Report as part of a quote request to Inel's AESMD. Closes #4

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ asm = ["xsave/asm"]
 
 [dependencies]
 xsave = { version = "0.1.1", default-features = false }
+serde = { version = "1.0", features = ["derive"] }
```
Comment (npmccallum) [inline Cargo.toml:31]: Make this optional.

File: src/types/attr.rs [hunk 2]
```diff
@@ -3,8 +3,11 @@
 //! Attributes (Section 38.7.1)
 //! The attributes of an enclave are specified by the struct below as described.
 
+use serde::{Deserialize, Serialize};
+
 bitflags::bitflags! {
     /// Section 38.7.1.
+    #[derive(Serialize, Deserialize)]
```
Comment (npmccallum) [inline src/types/attr.rs:10]: Indent is wrong

File: src/types/attr.rs [hunk 3]
```diff
@@ -31,6 +34,7 @@ impl Default for Flags {
 
 bitflags::bitflags! {
     /// Section 42.7.2.1; more info can be found at https://en.wikipedia.org/wiki/Control_register.
+    #[derive(Serialize, Deserialize)]
```
Comment (npmccallum) [inline src/types/attr.rs:37]: Indent is wrong

**Review by npmccallum — CHANGES_REQUESTED** [review]

File: src/types/attr.rs (hunk 2 above)
Comment (npmccallum) [reply src/types/attr.rs]: Err. Never mind. This was formatted poorly on my phone.

File: src/types/attr.rs (hunk 3 above)
Comment (npmccallum) [reply src/types/attr.rs]: Same as above.

**Review by npmccallum — APPROVED** [review]

**Review by connorkuehl — APPROVED** [review]

**Pull request merged**